common/devdoc/gballoc_hl_passthrough.md

```diff
@@ -0,0 +1,82 @@
+# gballoc_hl_passthrough requirements
+================
+
+## Overview
+
+gballoc_hl_passthrough is a module that delegates all call of its APIs to the ones from gballoc_ll. 

```

Comment:
>gballoc_hl_passthrough [](start = 0, length = 22)

So is this like an interface and there will be other implementations that do something besides passthrough? Sorry, not in the loop on all of this and trying to understand what the purpose of this layer is...

Reply:
We need/want an abstraction that "does nothing" for those times when we don't need to compute anything

And there's going to be another implementation that tracks metrics. 

A user is only supposed to be using the HL layer (which will become the gballoc we know).

---
In reply to: [458464922](https://github.com/Azure/azure-c-pal/pull/12#discussion_r458464922) [](ancestors = 458464922)